packages/components/addon/modifiers/hds-clipboard.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { assert } from '@ember/debug';
+
+export const getTextToCopy = (text) => {
+  let textToCopy;
+
+  if (text) {
+    if (typeof text === 'string') {
+      textToCopy = text;
+    } else if (
+      // context: https://github.com/hashicorp/design-system/pull/1564
+      typeof text === 'number' ||
+      typeof text === 'bigint'
+    ) {
+      textToCopy = text.toString();
+    } else {
+      assert(
+        `\`hds-clipboard\` modifier - \`text\` argument must be a string - provided: ${typeof text}`
+      );
+    }
+  }
+  return textToCopy;
+};
+
+export const getTargetElement = (target) => {
+  let targetElement;
+  if (typeof target === 'string') {
+    targetElement = document.querySelector(target);
+
+    if (!targetElement) {
+      console.error(
+        '`hds-clipboard` modifier - `target` selector provided does not point to an existing DOM node, check your selector string',
+        targetElement
+      );
+      return;
+    }
+  } else if (target instanceof Node && target.nodeType === 1) {
```

Comment:
for better readability we could write it as
```suggestion
  } else if (target instanceof Node && target.nodeType === Node.ELEMENT_NODE) {
```